internal/app/set/main_test.go

```diff
@@ -1,19 +1,42 @@
 package set
 
 import (
+	"fmt"
 	"testing"
 )
 
 func TestSetSerializeCommand(t *testing.T) {
-	t.Fatal("No tests written for serialize set command")
+	buffer1cmd, err := SerializeSetCommand("greeting", 0, 0, 13)
+	if err != nil {
+		e := fmt.Sprintf("An error occured where is shouldn't have: %v", err)
+		t.Fatal(e)
+	}
 
-	// Function should serialize some type of data (byte stream) to be sent to the memcached server
-	SerializeSetCommand()
-}
+	// expected size is 23 bytes => set greeting 0 0 13\r\n
+	if buffer1cmd.Len() != 23 {
+		e := fmt.Sprintf("Buffer size incorrect, got=%d, expected=%d", buffer1cmd.Len(), 23)
+		t.Fatal(e)
+	}
+
+	if buffer1cmd.String() != "set greeting 0 0 13\r\n" {
+		e := fmt.Sprintf("Incorrect buffer string, got=%s, expected=%s", buffer1cmd.String(), "set greeting 0 0 13\r\n")
+		t.Fatal(e)
+	}
+
+	buffer1dataBlock, err := SerializeSetDataBlock("Hello, World!")
```

Comment:
Instead of testing the `SerializeSetDataBlock` and the `SerilaizeSetCommand` in one test function, split them up into 2 separate ones.